dashboard/app/models/unit.rb

```diff
@@ -713,8 +713,9 @@ def link(unit_group_unit: nil)
   # @param user [User]
   # @param locale [String] User or request locale. Optional.
   # @return [Boolean] Whether the user can view the unit.
-  def can_view_version?(user, locale: nil)
-    return false unless Ability.new(user).can?(:read, self)
+  def can_view_version?(user, locale: nil, unit_group: nil)
+    unit_group ||= self.unit_group
```

Comment:
`self.unit_group` is deprecated, but  I think it's beyond the scope of the PR to `can_view_version?`

Reply:
good catch! I think `original_unit_group` is really what I want here.

Reply:
this turned out to be a can of worms, because `original_unit_group` is not in the script cache yet, so using it directly adds more queries. I don't want to add it to the cache in this PR, because it adds a fair amount of risk, and this PR is already quite big. 

I think the `self.unit_group` method actually does what I want, because it loads the unit group from the unit group cache, without adding more queries. perhaps the right fix is to undeprecate it and rename it to `get_original_unit_group`? (also for another PR)